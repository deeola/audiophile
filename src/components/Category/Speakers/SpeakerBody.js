import React from "react";
import ProductWithImage from "../shared/ProductWithImage";
import { SpeakerCont } from "./Speakerstyling";

import Z7desktop from "../../../assets/category-speakers/desktop/image-zx9.jpg";
import Z9desktop from "../../../assets/category-speakers/desktop/image-zx7.jpg";

import Z7tablet from "../../../assets/category-speakers/tablet/image-zx9.jpg";
import Z9tablet from "../../../assets/category-speakers/tablet/image-zx7.jpg";

import Z7mobile from "../../../assets/category-speakers/mobile/image-zx9.jpg";
import Z9mobile from "../../../assets/category-speakers/mobile/image-zx7.jpg";

const SpeakerBody = () => {
  // use inner width to determine image soure
  let Z9speaker = Z7desktop;
  let Z7speaker = Z9desktop;

  if (window.innerWidth < 802) {
    Z9speaker = Z7tablet;
    Z7speaker = Z9tablet;
  } else if (window.innerWidth < 490) {
    Z9speaker = Z7mobile;
    Z7speaker = Z9mobile;
  }

  return (
    <SpeakerCont data-testid='speaker'>
      <ProductWithImage
        link="/zx9"
        img={Z9speaker}
        label="SEE PRODUCT"
        np={"NEW PRODUCT"}
        title="ZX9 speakers"
        subtitle={
          "Upgrade your sound system with the all new ZX9 active speaker. Itâ€™s a bookshelf speaker system that offers truly wireless connectivity -- creating new possibilities for more pleasing and practical audio setups."
        }
      />
      <ProductWithImage
        primary
        link="/zx7"
        img={Z7speaker}
        label="SEE PRODUCT"
        title="ZX7 speaker"
        subtitle={
          " Stream high quality sound wirelessly with minimal loss. The ZX7 bookshelf speaker uses high-end audiophile components that represents the top of the line powered speakers for home or studio use."
        }
      />
    </SpeakerCont>
  );
};

export default SpeakerBody;
