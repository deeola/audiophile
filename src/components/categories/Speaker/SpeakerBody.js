import React from 'react'
import SpeakerSec from './SpeakerSec'


function SpeakerBody(props) {
    return (
        <div>
            <SpeakerSec />

        </div>
    )
}



export default SpeakerBody

