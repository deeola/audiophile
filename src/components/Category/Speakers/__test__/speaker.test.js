import { render, cleanup } from "@testing-library/react";
import ReactDOM from "react-dom";
import renderer from "react-test-renderer";
import { BrowserRouter } from "react-router-dom";
import SpeakerBody from "../SpeakerBody";

afterEach(cleanup);

it("renders speaker section correctly", () => {
  const section = document.createElement("section");
  ReactDOM.render(
    <BrowserRouter>
      <SpeakerBody />
    </BrowserRouter>,
    section
  );
});

it("contains two element children", () => {
  const { getByTestId } = render(
    <BrowserRouter>
      <SpeakerBody />
    </BrowserRouter>
  );

  expect(getByTestId('speaker').childElementCount).toBe(2)
});

// create snapshot

it('matches snapshot', () => {
    const tree  = renderer.create(
        <BrowserRouter>
        <SpeakerBody />
      </BrowserRouter>
    ).toJSON()

    expect(tree).toMatchSnapshot()
})