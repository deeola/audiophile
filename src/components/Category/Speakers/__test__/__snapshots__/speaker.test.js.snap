// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<section
  className="sc-papXJ bxEehH"
  data-testid="speaker"
>
  <div
    className="sc-bczRLJ gIwRId"
    data-testid="productimage"
  >
    <img
      alt="ZX9 speakers"
      className="sc-dkzDqf gRVFrI"
      src="image-zx9.jpg"
    />
    <article
      className="sc-hKMtZM bQLcuJ"
      data-testid="productbutton"
    >
      <div
        className="sc-eCYdqJ bMkDvt"
      >
        <p
          className="sc-iBkjds dIztKO"
          data-testid="np"
        >
           
          NEW PRODUCT
        </p>
        <h2
          className="sc-jSMfEi imCpJA"
          data-testid="title"
        >
          ZX9 speakers
        </h2>
        <p
          className="sc-gKXOVf dgecyH"
          data-testid="subtitle"
        >
          Upgrade your sound system with the all new ZX9 active speaker. Itâ€™s a bookshelf speaker system that offers truly wireless connectivity -- creating new possibilities for more pleasing and practical audio setups.
        </p>
      </div>
      <a
        className="sc-ftvSup KzLBC"
        color="primary"
        data-testid="productbutton-ZX9 speakers"
        hover="primary"
        href="/zx9"
        onClick={[Function]}
      >
        SEE PRODUCT
      </a>
    </article>
  </div>
  <div
    className="sc-bczRLJ htivJk"
    data-testid="productimage"
  >
    <img
      alt="ZX7 speaker"
      className="sc-dkzDqf gRVFrI"
      src="image-zx7.jpg"
    />
    <article
      className="sc-hKMtZM bQLcuJ"
      data-testid="productbutton"
    >
      <div
        className="sc-eCYdqJ bMkDvt"
      >
        <p
          className="sc-iBkjds dIztKO"
          data-testid="np"
        >
           
        </p>
        <h2
          className="sc-jSMfEi imCpJA"
          data-testid="title"
        >
          ZX7 speaker
        </h2>
        <p
          className="sc-gKXOVf dgecyH"
          data-testid="subtitle"
        >
           Stream high quality sound wirelessly with minimal loss. The ZX7 bookshelf speaker uses high-end audiophile components that represents the top of the line powered speakers for home or studio use.
        </p>
      </div>
      <a
        className="sc-ftvSup KzLBC"
        color="primary"
        data-testid="productbutton-ZX7 speaker"
        hover="primary"
        href="/zx7"
        onClick={[Function]}
      >
        SEE PRODUCT
      </a>
    </article>
  </div>
</section>
`;
